activerecord/lib/active_record/inheritance.rb

```diff
@@ -40,6 +40,8 @@ module Inheritance
       # This is true, by default.
       class_attribute :store_full_sti_class, instance_writer: false
       self.store_full_sti_class = true
+
+      @_type_candidates_cache = Concurrent::Map.new
```

Comment:
Actually anonymous classes can't be used because method itself needs class name to compute type, so I've used existing named classes.